### DISCUSSION TITLE
OpenSSL 1.1.1 lookup_sess_in_cache fails if Session version mismatch with ClientHello

### DISCUSSION THREAD

**Discussion by neovasant** [openssl/openssl#24653] [Q&A]
We are working on upgrading from OpenSSL 1.0.2k to 1.1.1 version. We use pre-created TLSV1.0 sessions for resumption. We notice that Session look up fails [(lookup_sess_in_cache)](https://github.com/openssl/openssl/blob/master/ssl/ssl_sess.c#L565) if ClientHello has a different version say TLS1.2 version. I guess LHash key might be using version as well. If this is correct, it seems TLS version consistency check that follows [here](https://github.com/openssl/openssl/blob/master/ssl/ssl_sess.c#L581) seems redundant. Can someone clarify this behaviour ? 
Apologies if this is not an issue.

**Comment by tom-cosgrove-arm** [comment]
Given the date on this is 2019, I presume this is an old issue converted to a discussion. However, 1.1.1 has been out of support for months now, so this is unlikely to be answered
